### DISCUSSION TITLE
RancidBacon's audiogif: How to convert MP4 to an a.gif?

### DISCUSSION THREAD

**Discussion by Klipperken** [community/community#147037] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Greetings:

My sincere apologies, bit I am not familiar with Unix/Linux OS, programs, or CLI.

Apparently, A.Gif can convert video to .a.gif (as in: MP4 to .a.gif), but I do not get this to work, possibly I do not have all the needed parts, though I think I do:
Python 3.13.1 (with Removed Path Length Limitations)
A.GIF was cloned.
Long ago, an FFMPG codec was installed.

As for Kaitai Struct Python Library,  ̶I̶ ̶c̶a̶n̶n̶o̶t̶ ̶f̶i̶n̶d̶ ̶a̶ ̶d̶o̶w̶n̶l̶o̶a̶d̶ ̶l̶i̶n̶k̶ ̶o̶r̶ ̶a̶n̶y̶t̶h̶i̶n̶g̶,̶ ̶t̶h̶e̶r̶e̶ ̶i̶s̶ ̶a̶ ̶c̶o̶m̶m̶a̶n̶d̶ ̶t̶h̶o̶u̶g̶h̶ ̶(̶O̶r̶,̶ ̶i̶t̶ ̶l̶o̶o̶k̶s̶ ̶l̶i̶k̶e̶ ̶o̶n̶e̶ ̶t̶o̶ ̶m̶e̶,̶ ̶b̶u̶t̶ ̶a̶s̶ ̶I̶ ̶s̶a̶i̶d̶ ̶.̶.̶.̶ ̶n̶o̶t̶ ̶f̶a̶m̶i̶l̶i̶a̶r̶ ̶w̶i̶t̶h̶ ̶U̶n̶i̶x̶/̶L̶i̶n̶u̶x̶)̶:̶
̶p̶i̶p̶ ̶i̶n̶s̶t̶a̶l̶l̶ ̶k̶a̶i̶t̶a̶i̶s̶t̶r̶u̶c̶t̶
̶I̶F̶ ̶t̶h̶i̶s̶ ̶i̶s̶ ̶s̶o̶m̶e̶ ̶i̶n̶s̶t̶a̶l̶l̶ ̶c̶o̶m̶m̶a̶n̶d̶ ̶.̶.̶.̶ ̶I̶ ̶h̶a̶v̶e̶ ̶n̶o̶ ̶i̶d̶e̶a̶ ̶w̶h̶e̶r̶e̶ ̶t̶o̶ ̶i̶n̶s̶e̶r̶t̶ ̶t̶h̶i̶s̶ ̶c̶o̶m̶m̶a̶n̶d̶ ̶o̶n̶ ̶P̶C̶.̶
Well, I did find the download link.
A TAR.GZ file (can be opened with many a tool (like 7z), but I do not even know, if this is a necessity. What do I do with this file? Point Git Desktop to it? Unzip it to a folder, and then point Git Desktop to it? Eat it? Try to knit with it (I will ask my wife then, she is the knitting one here.  ;) )?
While I am well known with Windows, I am hopeless in "Unixing"/"Linuxing".  :(

I saw these CLI lines:
ffmpeg -ss MM:SS.mm -t SS.mm -i input_video.mp4 -vf "crop=out_w=660:out_h=522:x=250:y=0" -c:a aac -map 0:v -map 0:a my_new.mp4
or:
ffmpeg -ss MM:SS.mm -t SS.mm -i input_video.mp4 -ar 22050 -ac 1 my_new.wav

And:
tools/gif-audio-add.py my_new.gif my_new.wav

Alas, in this whole I am entirely lost.

Can someone please help me out?

Thank you aforehand:
Klip.

EDIT:
If anyone knows a better, or more user-friendly (For PC-using simpletons like me) AudioGIF converter, please tell me.
Thank you:
Klip.

**Comment by KannaDev** [comment]
If it's more of a one-time thing, then probably something like [ezgif](https://ezgif.com/video-to-gif)'s mp4 to gif could get the job done.

**Reply by Klipperken** [reply]
AH, marvelous, thank you, good miss!
Most appreciated!!!

I will hop over there right away.

Thank you:
Klip.

**Reply by Klipperken** [reply]
Apologies, but the sound got killed over there, as well.
And I see no way to re-add the sound to the file?

Is it possible, you made a mistake, miss?
Or have I missed something?

**Reply by Klipperken** [reply]
Does anyone know where to make high-quality GIF with sound, please?
